pkg/query-service/app/inframetrics/pods.go

```diff
@@ -0,0 +1,386 @@
+package inframetrics
+
+import (
+	"context"
+	"sort"
+
+	"go.signoz.io/signoz/pkg/query-service/app/metrics/v4/helpers"
+	"go.signoz.io/signoz/pkg/query-service/common"
+	"go.signoz.io/signoz/pkg/query-service/interfaces"
+	"go.signoz.io/signoz/pkg/query-service/model"
+	v3 "go.signoz.io/signoz/pkg/query-service/model/v3"
+	"go.signoz.io/signoz/pkg/query-service/postprocess"
+	"golang.org/x/exp/slices"
+)
+
+var (
+	metricToUseForPods = "k8s_pod_cpu_utilization"
+
+	podAttrsToEnrich = []string{
+		"k8s_pod_uid",
+		"k8s_pod_name",
+		"k8s_namespace_name",
+		"k8s_node_name",
+		"k8s_deployment_name",
+		"k8s_statefulset_name",
+		"k8s_daemonset_name",
+		"k8s_job_name",
+		"k8s_cronjob_name",
+	}
+
+	k8sPodUIDAttrKey = "k8s_pod_uid"
+
+	queryNamesForPods = map[string][]string{
+		"cpu":            {"A"},
+		"cpu_request":    {"B", "A"},
+		"cpu_limit":      {"C", "A"},
+		"memory":         {"D"},
+		"memory_request": {"E", "D"},
+		"memory_limit":   {"F", "D"},
+		"restarts":       {"G", "A"},
+	}
+	podQueryNames = []string{"A", "B", "C", "D", "E", "F", "G"}
+
+	metricNamesForPods = map[string]string{
+		"cpu":            "k8s_pod_cpu_utilization",
+		"cpu_request":    "k8s_pod_cpu_request_utilization",
+		"cpu_limit":      "k8s_pod_cpu_limit_utilization",
+		"memory":         "k8s_pod_memory_usage",
+		"memory_request": "k8s_pod_memory_request_utilization",
+		"memory_limit":   "k8s_pod_memory_limit_utilization",
+		"restarts":       "k8s_container_restarts",
```

Comment:
same comment, there would be many metrics, I think it's better to put them inside a `metrics_constants.go` or somethings similar, so that they can be used elsewhere.